docs/specification.rst

```diff
@@ -1,668 +1,481 @@
+------------------------------------
+
+At this point in time, ``Intersection`` as a value expression is only
+consistent with a type expression of ``object``, ``Any``, and a few internal
+types that should not be publically used.
+
+This is not a binding limitation on Intersection, and future PEPs which provide
+ways to type methods which do runtime introspection should treat this the same
+as other parameterizable type forms which exist to express typing concepts
+and not to express a specific indivudal runtime type.
+
+
+Optional checks which are not strictly about correctness
+========================================================
+
+Type checkers and/or linters may decide to provide opinionated rules.
+A few anticipated ones are listed below.
+
+Type checkers and linters MAY provide any of these or others but MUST NOT
+use these rules to treat libraries which have not opted into these checks as
+being in-error. The cases these detect have validity, but also have sharp edges
+that some users may choose to want to avoid in their own code.
+
+
+Allowing avoiding unintentional ambiguity with gradual types
+------------------------------------------------------------
+
+- Users may wish to ban ``Any`` or gradual types in intsersections.
+- Users may wish to enforce that Gradual types are ordered after
+  non-gradual types in intersections
+- Detecting cases where non-disjoint use could introduce ordering concerns
+  (See appendix below for one method of detecting potential ordering concerns)
+
+Rationale
+=========
+
+Outside of the fact that we are specifying the intersections should be ordered
+in python, very little about this should be surprising. The ordering allows
+for allowing developer choice in resolving ambiguity, and comes with
+a side-effect of allowing linear complexity when evaluating
+intersections by type checkers
+
+As ordering of an intersection has the potential to be controversial,
+the below summarizes what was explored and the various tradeoffs
+
+Ordering
+--------
+
+It was observed that by introducing an ordering on one direction of the
+consistency checks, that for all of the anticipated cases involving fully typed
+code, the desired behavior from the type system for motivating cases worked as
+intended, and that for the case of ``Any & T``, it matched the behavior of
+subclassing of ``Any``, and that any issue of LSP violating types was already
+handled by the errors one would get if attempting to create an invalid type.
+
+While the pure and unordered form would be identical for the vast majority of
+cases we expect users of fully typed code to encounter, The combination of the
+definition of  ``Any`` doubling for both compatability and uncertainty creates
+a situation where it cannot be excluded from such an intersection, and that
+none of the possible interpretations of it are likely to be satisfying for all
+users.
+
+There are a few potential unordered interpretations of ``Any & T``

```

Comment:
Wondering if we should only include the one we're proposing in the PEP - or maybe put the others in "rejected ideas"?

Reply:
Waiting to summarize each and why not for each of those until the pending discussion. It may change how compatible or not those options are.